packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -206,15 +213,57 @@ void diagnoseXcodeBuildFailure(XcodeBuildResult result) {
     printError("  open ios/Runner.xcodeproj");
     return;
   }
+  if (result.buildForDevice
+      && result.buildCommands != null
+      && result.buildCommands.contains('build')
+      && result.appDirectory != null) {
```

Comment:
Some grouping would indeed be nicer. Pulled out an inner class and used assertions.